dmriprep/utils/vectors.py

```diff
@@ -1,482 +1,346 @@
 
-    # Round bvals by bmag.
-    bval_mag_norm = round_bvals(bvals, bmag=bmag)
+    >>> affine = np.eye(4)
+    >>> affine[0, 0] *= -1.0  # Make it LAS
+    >>> bvecs2ras(affine, [(1.0, 0.0, 0.0), (-1.0, 0.0, 0.0)]).tolist()
+    [[-1.0, 0.0, 0.0], [1.0, 0.0, 0.0]]
 
-    # Rescale bvecs.
-    bvec_rescaled = rescale_bvec(bvecs, B0_NORM_EPSILON)
-    return bval_mag_norm, bvec_rescaled
+    >>> affine = np.eye(3)
+    >>> affine[:2, :2] *= -1.0  # Make it LPS
+    >>> bvecs2ras(affine, [(1.0, 0.0, 0.0), (-1.0, 0.0, 0.0)]).tolist()
+    [[-1.0, 0.0, 0.0], [1.0, 0.0, 0.0]]
 
+    >>> affine[:2, :2] = [[0.0, 1.0], [1.0, 0.0]]  # Make it ARS
+    >>> bvecs2ras(affine, [(1.0, 0.0, 0.0), (-1.0, 0.0, 0.0)]).tolist()
+    [[0.0, 1.0, 0.0], [0.0, -1.0, 0.0]]
 
-class VectorTools(object):
-    """
-    Class of vector-handling methods for io and integrity checking.
+    >>> bvecs2ras(affine, [(0.0, 0.0, 0.0)]).tolist()
+    [[0.0, 0.0, 0.0]]
 
-    Parameters
-    ----------
-    basedir : str
-        Path the output base directory for derivative dmriprep data.
-    dwi_file : str
-        Optionally provide a file path to the diffusion-weighted image series to which the
-        bvecs/bvals should correspond.
-    fbval : str
-        File path of the b-values.
-    fbvec : str
-        File path of the b-vectors.
-    rasb_tsv_file : pathlike or str
-        Path to .tsv file containing RAS-B gradient table.
-    B0_NORM_EPSILON : float
-        Gradient threshold below which volumes and vectors are considered B0's. Default is 50.
-    bval_scaling : bool
-        If True, then normalizes b-val by the square of the vector amplitude. Default is True.
     """
-    def __init__(self, basedir, dwi_file, fbval=None, fbvec=None, rasb_tsv_file=None, B0_NORM_EPSILON=50,
-                 bval_scaling=True):
-        from pathlib import Path
-        self.dwi_file = dwi_file
-        self.fbval = fbval
-        self.fbvec = fbvec
-        self.basedir = basedir
-        self.dwi_file = dwi_file
-        self.B0_NORM_EPSILON = B0_NORM_EPSILON
-        self.bval_scaling = bval_scaling
-        self.rasb_tsv_file = rasb_tsv_file
-        self.basedir = Path(self.basedir)
-        self.bvals = None
-        self.bvecs = None
-        self.bmag = None
-        self.gtab = None
-        self.b0_ixs = None
-
-    def read_bvals_bvecs(self):
-        # Read bvec/bval files
-        from pathlib import Path
-        from dipy.io import read_bvals_bvecs
-
-        # Read b-values and b-vectors from disk Files can be either ‘.bvals’/’.bvecs’ or ‘.txt’ or ‘.npy’ (containing
-        # arrays stored with the appropriate values).
-
-        if Path(self.fbval).exists() and Path(self.fbvec).exists():
-            self.bvals, self.bvecs = read_bvals_bvecs(self.fbval, self.fbvec)
-        else:
-            raise OSError("Either bval or bvec files not found!")
-
-        # Get b order of magnitude
-        self.bmag = int(np.log10(np.max(self.bvals))) - 1
```

Comment:
this is the default value of dipy's round_bvals - if you're not going to change that behavior just don't overload it (it's quite risky!)